pkg/model/crd.go

```diff
@@ -657,6 +657,45 @@ func (r *CRD) GetAllRenames(op OpType) (map[string]string, error) {
 	return renames, nil
 }
 
+// FindIdentifierInShape returns true and the resource's identifier iff shape
+// contains a matching field or corresponding plural meaning it can be populated
+// using the resource identifier.
+func (r *CRD) FindIdentifierInShape(
+	shape *awssdkmodel.Shape) (string, bool) {
+	resIdentifier := ""
+	found := false
+	resourceIdentifiers := []string{
+		"Name",
+		"Names",
+		r.Names.Original + "Name",
+		r.Names.Original + "Names",
+		r.Names.Original + "Id",
+		r.Names.Original + "Ids",
+	}
+
+	// Locate an identifier within resource
```

Comment:
Some resources may have multiple identifiers :D 

Example: An [APIGateway API](https://docs.aws.amazon.com/sdk-for-go/api/service/apigatewayv2/#GetApiOutput) has `ApiId` and `Name`. Therefore we will have `Spec.Name` and `Status.APIID`. 

We won't know ahead of time which identifier will be needed in the `ReadMany` filter. Lets create an array of identifiers and then loop over them in L690?

Reply:
Good callout. I'm refactoring this logic into 2 separate helper funcs (will be in latest revision), and I think you're right that I'll need to return a slice of identifiers:

If the scenario happens where the Shape contains `Names` and `Ids` field, but the CRD only contains `Id` field, then returning just the Names and comparing Id would yield no required fields (when really we want to require `Id` because `Ids` is present in Shape).